app/uk/gov/hmrc/digitalservicestax/aa_data/JsonProtocol.scala

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.digitalservicestax.aa_data
+
+import enumeratum.EnumFormats
+import play.api.libs.json._
+import uk.gov.hmrc.digitalservicestax.data._
+
+object JsonProtocol {
+
+  implicit val nonEmptyStringFormat: Format[NonEmptyString] = new Format[NonEmptyString] {
+    override def reads(json: JsValue): JsResult[NonEmptyString] = json match {
+      case JsString(value) if value.length > 0 => JsSuccess(NonEmptyString.apply(value))
```

Comment:
I'm being massively pedantic here, but `if value.nonEmpty` would be more efficient.